open-apparel-registry/open-apparel-registry
# Create a fixture to load a trained model in development

## Overview

This PR creates fixtures to populate the api_trainedmodel and dedupe_indexed_records tables to avoid training a model in the dev environment. It also updates GazetteerCache to load the saved model instead of training a model when django starts up.

Connects #2101

## Testing Instructions

* run `./scripts/resetdb`
* run `./scripts/dbshell`
* run the query below. There should be two saved models, the second of which is active.
```
openapparelregistry=# select id, created_at,is_active,activated_at from api_trainedmodel;
 id |         created_at         | is_active |        activated_at
----+----------------------------+-----------+----------------------------
  1 | 2022-09-06 20:52:21.913+00 | f         |
  2 | 2022-09-06 20:54:45.517+00 | t         | 2022-09-06 21:02:51.454+00
(2 rows)
```
* run the query below to ensure there is data in the `dedupe_indexed_records` table
```
openapparelregistry=# select * from dedupe_indexed_records limit 3;
 id | block_key |    record_id    |                                                                        rec
ord_data
----+-----------+-----------------+---------------------------------------------------------------------------
--------------------------------------------------------------------------------
  1 | wenzhou:0 | CN20222495E9C9Q | {"country": "cn", "name": "wenzhou jietu(aolun) shoes factory", "address":
 "no.8 shuangbao west road ouhai economic developm p.c. 325014 wenzhou china"}
  2 | no8sh:1   | CN20222495E9C9Q | {"country": "cn", "name": "wenzhou jietu(aolun) shoes factory", "address":
 "no.8 shuangbao west road ouhai economic developm p.c. 325014 wenzhou china"}
  3 | quanzho:0 | CN20222491QG6M8 | {"country": "cn", "name": "quanzhou xinxing footwear co. ltd", "address":
"wuyi industial zone baiqi village huian countyquanzhou city fujian province#"}
(3 rows)
```


## Checklist
